UOP1_Project/Assets/Scripts/Systems/Settings/SettingsSystemGraphicsComponent.cs

```diff
@@ -0,0 +1,208 @@
+        //RESOLUTION
+        //TODO: apparently, unity keeps the resolution which was set in the last session. But there is no method to get this resolution in the next session
+        //TODO: because Screen.currentResolution in windowed mode gives back the resolution of the desktop, not the unity window resolution
+        currentResolution = Screen.resolutions[Screen.resolutions.Length - 1]; //Initially set highest possible resolution on device
+        previousResolution = currentResolution;
+        Screen.SetResolution(currentResolution.width, currentResolution.height, Screen.fullScreenMode, currentResolution.refreshRate);
+        SelectCurrentResolution();
+        
+        //FULLSCREEN MODE
+        //TODO: load previous session fullscreen setting
+        FullScreen = Screen.fullScreenMode == FullScreenMode.ExclusiveFullScreen;
+        previouslyInFullScreen = FullScreen;
+        fullscreenToggle.SetIsOnWithoutNotify(FullScreen);
+        
+        //QUALITY PRESET
+        //TODO: instead of hardcoding the level here, load data from previous session
+        currentQualityIndex = (int) SettingsPresetsScriptableObject.GraphicsQualityLevel.Low;
+        previousQualityIndex = currentQualityIndex;
+        currentAdvancedGraphics = settingsPresets.presetList[currentQualityIndex]; //Set to lowest preset initially
+        previousAdvancedGraphics = currentAdvancedGraphics;
+        graphicsPresetsDropdown.SetValueWithoutNotify(currentAdvancedGraphics.custom ? graphicsPresetsDropdown.options.Count-1 : currentQualityIndex);
+        UpdateUI();
+        setupComplete = true;
+    }
+    
+    List<TMP_Dropdown.OptionData> GetResolutionsDropdownData()
+    {
+        List<TMP_Dropdown.OptionData> options = new List<TMP_Dropdown.OptionData>();
+        for (int i = 0; i < Screen.resolutions.Length; i++)
+        {
+            options.Add(new TMP_Dropdown.OptionData(Screen.resolutions[i].ToString()));
+        }
+
+        return options;
+    }
+    
+    void UpdateUI()
+    {
+        anisotropicFilteringDropdown.SetValueWithoutNotify((int) currentAdvancedGraphics.anisotropicFiltering);
+        antiAliasingSlider.SetValueWithoutNotify(currentAdvancedGraphics.antiAliasing);
+        shadowDistanceSlider.SetValueWithoutNotify(currentAdvancedGraphics.shadowDistance);
+        shadowQualityDropdown.SetValueWithoutNotify((int) currentAdvancedGraphics.shadowQuality);
+        shadowDistanceText.text = currentAdvancedGraphics.shadowDistance.ToString();
+        antiAliasingText.text = currentAdvancedGraphics.antiAliasing.ToString();
+        fullscreenToggle.SetIsOnWithoutNotify(FullScreen);
+        graphicsPresetsDropdown.SetValueWithoutNotify(currentAdvancedGraphics.custom ? graphicsPresetsDropdown.options.Count-1 : currentQualityIndex);
+    }
+
+    #region UI CALLBACKS
+
+    public void OnChangeFullscreen(bool fullScreen)
+    {
+        Screen.fullScreenMode = fullScreen ? FullScreenMode.ExclusiveFullScreen : FullScreenMode.Windowed;
```

Comment:
Unity seems to support Borderless Windowed mode aswell (see FullScreenWindow in [FullScreenMode](https://docs.unity3d.com/ScriptReference/FullScreenMode.html)). I don't know if this option is already defined to be just a toggle, but we might want to support this option later.